charts/kubernetes-agent/README.md

```diff
@@ -42,14 +42,37 @@ A Helm chart for the Octopus Kubernetes Agent
 | agent.serverSubscriptionId | string | `""` | The subscription ID that is used to by the agent to identify itself with Octopus Server |
 | agent.serverUrl | string | `""` | The URL of the target Octopus Server to register this agent with |
 | agent.serviceAccount.annotations | object | `{}` | Annotations to add to the autogenerated service account |
-| agent.serviceAccount.name | string | Generates a name based on `targetName` | The name of the service account for the agent pod |
+| agent.serviceAccount.name | string | Generates a name based on `agent.name` | The name of the service account for the agent pod |
 | agent.space | string | `"Default"` | The Space to register the agent in |
-| agent.targetEnvironments | list | `[]` | The target environments to register the agent with |
-| agent.targetName | string | `""` | The name of the deployment target |
-| agent.targetRoles | list | `[]` | The target roles to register the agent with |
-| agent.targetTenantTags | list | `[]` | The target tenant tags to register the agent with |
-| agent.targetTenantedDeploymentParticipation | string | `"Untenanted"` | Can be `Untenanted`, `TenantedOrUntenanted` or `Tenanted`. |
-| agent.targetTenants | list | `[]` | The target tenants to register the agent with |
+
+### Agent as Deployment Target values
+
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| agent.deploymentTarget.enabled | bool | `false` | Set to register the agent as a Deployment Target using the provided initial values |
+
+### Agent as Deployment Target initial values
+
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| agent.deploymentTarget.initial.defaultNamespace | string | `""` | The default Kubernetes namespace for deployments |
+| agent.deploymentTarget.initial.environments | list | `[]` | The deployment target environments to register the agent with |
+| agent.deploymentTarget.initial.tags | list | `[]` | The deployment target tags to register the agent with |
+| agent.deploymentTarget.initial.tenantTags | list | `[]` | The deployment target tenant tags to register the agent with |
+| agent.deploymentTarget.initial.tenantedDeploymentParticipation | string | `"Untenanted"` | Can be `Untenanted`, `TenantedOrUntenanted` or `Tenanted`. |
+| agent.deploymentTarget.initial.tenants | list | `[]` | The deployment target tenants to register the agent with |
+
+### Agent as Worker values
+
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| agent.worker.enabled | bool | `false` | Set to register the agent as a Worker using the provided initial values |
+
+### Agent as Worker initial values
+
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| agent.worker.initial.workerPools | list | `[]` | The worker pools to associate with the worker |
```

Comment:
do we need to mention that its a list of _strings_ - or is that implicit?

Reply:
I believe that is implicit. The doc generator tool produces this based on the default value in the values.yaml. The default there is simply `[]` which in helm/yaml terms can mean any type.